Allow table sorting

loganmzz opened the pull request:
It is a PR about [Bugzilla ID 52962](https://bz.apache.org/bugzilla/show_bug.cgi?id=52962)

PR is cut into many commits for each required step for full implementation. Final commit enables sorting for aggregate report, view results in table and summary report.

Any table relying on `ObjectTableModel` must be supported.

loganmzz committed:
Bug 52962 - Fixes fired event row indexes

Last row index is INCLUSIVE ! It causes IndexOutOfBounds when using
table sorter.

loganmzz committed:
Bug 52962 - Adds sorter based on ObjectTableModel

loganmzz committed:
Bug 52962 - Adds table sorting

Available on components: aggregate report, view results in table and
summary report

pmouawad commented:
Hello,
Would it be possible to provide a version that has no conflict with TableVisualizer.java ?

There was some work on Sonar quality report, but I'll pause it on my side until this PR is merged if rest of the team is ok with merging.

Thanks

loganmzz committed:
Bug 52962 - Comments table model event backer API

loganmzz committed:
Bug 52962 - Relies on default table header renderer

It makes possible to use fully featured table header such as sort icon
support.

loganmzz commented:
I have commented TableModelEventBacker, and rebase it on trunk: d29ac213d3971942c8d98c4ef7223fed97001cfc (Sonar : fix squid:UselessParenthesesCheck Remove those useless parentheses.)

You must force fetch. Or delete remote branch references and refetch it.

pmouawad commented:
Hello,
I have commited part of the PR and made some changes:
- I didn't want to use everywhere HeaderAsPropertyRenderer.install as it removes formatting without introducing sorting
- It would have also broken backward compatibility on the class potentially breaking backward compatibility of plugins
- So I have replaced it by a HeaderAsPropertyRendererWrapper class
- I have added sorting to Aggregate Graph
- In ObjectTableModel, I have also replaced the modified method by a new one getObjectListAsList to avoid breaking backward compatibility

Thanks for your PR, please test nightly build to see if it works fine for you.
Thanks
